### ISSUE TITLE
Decide whether to specify exactly when a session is closed.

### ISSUE THREAD

**Issue by rgwilton** [netmod-wg/yang-ver-dt#40]
If a server supports multiple schema-sets in parallel, but a client removes a schema-set that is in use on a NETCONF session from the 'selectable' leaf-list, then TBD:

1. any NETCONF sessions that were using that schema-set are disconnected, or
2. any NETCONF sessions that were using that schema-set remain connected, but no new NETCONF sessions can use the schema-set that was removed from the 'selectable' leaf-list. If the server can only support a limited number of schema-sets in parallel, and the maximum number of schema-sets are currently in use, and a new NETCONF session attempts to use a new valid schema-set (exceeding the maximum number of parallel schema-sets), then TBD (disconnect those old NETCONF sessions that are using a schema-set that was removed from 'selectable', or block the new NETCONF session?)

**Comment by xorrkaz** [comment]
Given that this solution adds additional complexity, I think we should opt for simpler.  I prefer option 1.  Either that or we consider an option 3 whereby the server rejects the config if there are active client sessions.

**Comment by jsterne** [comment]
I'd prefer #1 as well.  Here is an expanded description to cover a few cases:

If a server supports multiple schema-sets in parallel, but a client removes a schema-set that is in use on a NETCONF session from the 'selectable' leaf-list, then any NETCONF sessions that were using the removed schema-set are disconnected.

If the server can only support a limited number of schema-sets in parallel, and the maximum number of schema-sets are currently in use, and a new NETCONF session attempts to use a new valid schema-set (exceeding the maximum number of parallel schema-sets), then the new NETCONF session is disconnected.

Servers SHOULD avoid a situation whereby a new NETCONF session is unable to connect without specifying a schema-set (i.e. connecting and using the default schema-set SHOULD always be possible).

**Comment by Reshad-Rahman** [comment]
Out of the 2, I also prefer option 1 as well. 

Did we discuss preventing  a schema-set that is in use from being removed?

**Comment by jsterne** [comment]
I don't think so but we could touch base on that in the call? My initial reaction is that I'd prefer to avoid that. I think it might be better to allow config to always be changed (not be blocked by "state"). An admin should probably not be prevented from removing a schema-set if they really want to.

**Comment by xorrkaz** [comment]
> Out of the 2, I also prefer option 1 as well.
> 
> Did we discuss preventing a schema-set that is in use from being removed?

No.  I added it when doing the slides as I thought it was another option to consider.  I think that a session is transient and thus not a great choice to reject the config, but I figured I'd raise it.

**Comment by Reshad-Rahman** [comment]
Joe/Jason, I agree. Just thought it'd be good to track as discussed, didn't reailze it was on the slides.
